birdyboard.py

```diff
@@ -0,0 +1,505 @@
+            self.unlogged_in_menu_next_step()
+        elif next_step == "2":  # view public threads.
+            self.deserialize_chirps_library()
+            self.public_or_private = "public"
+            self.view_chirps()
+            self.view_chirps_next_step()
+        elif next_step == "3":  # view private threads.
+            self.public_or_private = "private"
+            self.view_chirps_next_step()
+        elif next_step == "4":  # new public thread.
+            self.public_or_private = "public"
+            self.new_chirp_menu()
+        elif next_step == "5":  # new private thread.
+            self.public_or_private = "private"
+            self.new_chirp_menu()
+        elif next_step == "x":  # exit.
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.logged_in_menu_next_step()
+
+# ###############################
+# ######## CREATE A USER ########
+# ###############################
+
+    def create_a_user_menu(self):
+        self.deserialize_users()
+        user_name = input("user name: ")
+        if user_name == "b":
+            print("going back")
+            print(self.unlogged_in_menu)
+            self.unlogged_in_menu_next_step()
+        elif user_name == "x":
+            print("goodbye.")
+            exit()
+        else:
+            for user in self.users:
+                if user_name == user["user_name"]:
+                    print("User name already taken!")
+                    self.what_if_user_name_is_taken()
+            password = input("password: ")
+            self.add_new_user(user_name, password)
+            print("logging in " + user_name + ".")
+            print(self.logged_in_menu)
+            self.logged_in_menu_next_step()
+
+    def what_if_user_name_is_taken(self):
+        print("'b' to go back.\n'x' to exit.\n1. Choose from a list of created users.\n2. Try creating a user with a different name.")
+        next_step = input(">> ")
+        if next_step == "1":
+            self.deserialize_users()
+            self.view_users_menu()
+            self.users_menu_next_step()
+        elif next_step == "2":
+            self.create_a_user_menu()
+        elif next_step == "b":
+            print("going back.")
+            print(self.unlogged_in_menu)
+            self.unlogged_in_menu_next_step()
+        elif next_step == "x":
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.what_if_user_name_is_taken()
+
+    def add_new_user(self, user_name, password):
+        self.deserialize_users()
+        self.users.append({"user_name": user_name, "password": password})
+        self.serialize_users()
+        self.user_name = user_name
+
+
+# ############################
+# ######## VIEW USERS ########
+# ############################
+# I am fully aware that this is a completely insecure way to handle a password system.
```

Comment:
It's so insecure and such bad coding practice to store passwords that I'd strongly suggest removing password functionality entirely.  No authentication at all is better than bad authentication.